src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/MakeOfferTest.java

```diff
@@ -0,0 +1,106 @@
+import uk.gov.hmcts.cmc.claimstore.BaseTest;
+import uk.gov.hmcts.cmc.claimstore.controllers.utils.sampledata.SampleClaim;
+import uk.gov.hmcts.cmc.claimstore.exceptions.IllegalSettlementStatementException;
+import uk.gov.hmcts.cmc.claimstore.models.Claim;
+import uk.gov.hmcts.cmc.claimstore.models.offers.MadeBy;
+import uk.gov.hmcts.cmc.claimstore.models.offers.Offer;
+import uk.gov.hmcts.cmc.claimstore.models.sampledata.offers.SampleOffer;
+import uk.gov.hmcts.cmc.claimstore.services.OffersService;
+import uk.gov.hmcts.cmc.claimstore.services.notifications.fixtures.SampleUserDetails;
+
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class MakeOfferTest extends BaseTest {
+
+    private static final Long CLAIM_ID = 12344321L;
+    private static final Long DEFENDANT_ID = 43211234L;
+    private static final String AUTH_TOKEN = "authDataString";
+
+    @SpyBean
+    private OffersService offersService;
+
+    @Before
+    public void beforeEachTest() {
+        when(claimRepository.getById(CLAIM_ID)).thenReturn(Optional.of(
+            SampleClaim.builder()
+                .withDefendantId(DEFENDANT_ID)
+                .build()
+        ));
+
+        when(userService.getUserDetails(AUTH_TOKEN)).thenReturn(
+            SampleUserDetails.builder()
+                .withUserId(DEFENDANT_ID)
+                .build()
+        );
+    }
+
+    @Test
+    public void shouldAcceptValidOfferByDefendantAndReturnCreatedStatus() throws Exception {
+        makeOffer(SampleOffer.validDefaults(), MadeBy.DEFENDANT.name())
+            .andExpect(status().isCreated())
+            .andReturn();
+    }
+
+    @Test
+    public void shouldReturnBadRequestIfPartyIsIncorrectlySpecified() throws Exception {
+        makeOffer(SampleOffer.validDefaults(), "I'm not a valid enum value")
+            .andExpect(status().isBadRequest())
+            .andReturn();
+    }
+
+    @Test
+    public void shouldReturnForbiddenIfUserIsNotPartyOnClaim() throws Exception {
+        when(userService.getUserDetails(AUTH_TOKEN)).thenReturn(
+            SampleUserDetails.builder()
+                .withUserId(-300L)
+                .build()
+        );
+
+        makeOffer(SampleOffer.validDefaults(), MadeBy.DEFENDANT.name())
+            .andExpect(status().isForbidden())
+            .andReturn();
+    }
+
+    @Test
+    public void shouldReturnForbiddenIfUserIsPartyOnClaimButClaimsToBeOppositeParty() throws Exception {
+        makeOffer(SampleOffer.validDefaults(), MadeBy.CLAIMANT.name())
+            .andExpect(status().isForbidden())
+            .andReturn();
```

Comment:
I was quite confused at first. Perhaps `makeOffer` could be renamed to `makeOfferAsDefendant`?

Reply:
`MadeBy` is passed there as a parameter.

Reply:
it is, but as a param for url, not user making the request